crates/linker/src/lib.rs

```diff
@@ -32,33 +32,35 @@ pub enum LinkType {
 }
 
 pub fn supported(link_type: LinkType, target: &Triple) -> bool {
-    if let LinkType::Executable = link_type {
-        match target {
-            Triple {
-                architecture: target_lexicon::Architecture::X86_64,
-                operating_system: target_lexicon::OperatingSystem::Linux,
-                binary_format: target_lexicon::BinaryFormat::Elf,
-                ..
-            } => true,
-
-            // macho support is incomplete
-            Triple {
-                operating_system: target_lexicon::OperatingSystem::Darwin,
-                binary_format: target_lexicon::BinaryFormat::Macho,
-                ..
-            } => false,
-
-            Triple {
-                architecture: target_lexicon::Architecture::X86_64,
-                operating_system: target_lexicon::OperatingSystem::Windows,
-                binary_format: target_lexicon::BinaryFormat::Coff,
-                ..
-            } => true,
-
-            _ => false,
+    match link_type {
+        LinkType::Executable => {
+            match target {
+                Triple {
+                    architecture: target_lexicon::Architecture::X86_64,
+                    operating_system: target_lexicon::OperatingSystem::Linux,
+                    binary_format: target_lexicon::BinaryFormat::Elf,
+                    ..
+                } => true,
+
+                // macho support is incomplete
+                Triple {
+                    operating_system: target_lexicon::OperatingSystem::Darwin,
+                    binary_format: target_lexicon::BinaryFormat::Macho,
+                    ..
+                } => false,
+
+                Triple {
+                    architecture: target_lexicon::Architecture::X86_64,
+                    operating_system: target_lexicon::OperatingSystem::Windows,
+                    binary_format: target_lexicon::BinaryFormat::Coff,
+                    ..
+                } => true,
+
+                _ => false,
+            }
         }
-    } else {
-        false
+        LinkType::Dylib => false,
```

Comment:
linking the dev tests (using zig, today)  into a dylib is slow. I think (and fear) that we need to go do this ourselves.

Reply:
maybe it's not so bad? it's just a fancy copy https://github.com/gimli-rs/object/blob/master/crates/examples/src/bin/pecopy.rs 

fancy because we need to join two files, the test code and the builtins.